backend/src/org/commcare/suite/model/Suite.java

```diff
@@ -77,6 +82,32 @@ public void setID(int ID) {
     public Vector<Menu> getMenus() {
         return menus;
     }
+
+    /**
+     * @return This suite's menus, limited to those
+     * relevant in the given EvaluationContext. 
+     */
+    public Vector<Menu> getMenus(EvaluationContext context) {
+        Vector<Menu> relevant = new Vector<Menu>();
+        for(Menu m : menus) {
+            String condition = m.getRelevancyCondition();
+            if (condition == null) {
+                relevant.addElement(m);
+            }
+            else {
+                XPathExpression parsed;
+                try {
+                    parsed = XPathParseTool.parseXPath(condition);
```

Comment:
this might have performance implications if we do it frequently. Parsing XPath expressions is a super costly operation. We solve this for the other commands by moving the parse code into the menu object itself and letting it soft cache the parsed out version. Would be good to replicate that process here if possible.